### ISSUE TITLE
[ES6] Inline argument type for arrow function requires parentheses

### ISSUE THREAD

**Issue by calebegg** [google/closure-compiler#1309]
Inline (`/** Foo */`-style) type annotations are ignored for single-argument arrow functions unless the (normally optional) parentheses are included around the argument.

See [demo](http://closure-compiler-debugger.appspot.com/#input0%3Dlet%2520foo%2520%253D%2520%252F**%2520number%2520*%252F%2520x%2520%253D%253E%2520x%2520*%25202%253B%250Afoo('a')%253B%250A%250Alet%2520bar%2520%253D%2520(%252F**%2520number%2520*%252F%2520x)%2520%253D%253E%2520x%2520*%25202%253B%250Abar('a')%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1)

**Comment by concavelenz** [comment]
@Dominator008

**Comment by Dominator008** [comment]
This is a bit ambiguous. In the example:
`let foo = /** number */ x => x * 2;`
The `/** number */` JSDocInfo gets attached on the FUNCTION node, which one can argue is correct.

**Comment by MatrixFrog** [comment]
I agree, I think the way it is now is fine (and is well documented internally though admittedly not externally).

**Comment by calebegg** [comment]
That's fair. I do wish there was some sort of "this inline annotation has no effect here" warning. I've made this mistake several times, and I've corrected it several times in code reviews.
